File: sicpa-web/src/app/models/role-user-dto.ts
import { GenericDto } from "./generic-dto";
import { RoleDto } from "./role-dto";
import { UserDto } from "./user-dto";

export class RoleUserDto extends GenericDto {

  id: number;
  role: RoleDto;
  user: UserDto;

  constructor() {
    super();
  }
}

File: sicpa-root/sicpa-dto/src/main/java/com/sicpa/demo/dto/RoleUserDto.java
package com.sicpa.demo.dto;

import java.io.Serializable;

import javax.validation.constraints.NotNull;

import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
public class RoleUserDto extends BaseDto implements Serializable {
	private static final long serialVersionUID = 1L;

	private Long id;

	@NotNull
	private RoleDto role;

	@NotNull
	private UserDto user;
}